orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -70,20 +70,19 @@ public TypeDescription type() {
   private static final String ICEBERG_ID_ATTRIBUTE = "iceberg.id";
   private static final String ICEBERG_REQUIRED_ATTRIBUTE = "iceberg.required";
 
-  private static final String ICEBERG_BINARY_TYPE_ATTRIBUTE = "iceberg.binary-type";
-  private static final String ICEBERG_INTEGER_TYPE_ATTRIBUTE = "iceberg.integer-type";
-  private static final String ICEBERG_FIELD_LENGTH = "iceberg.length";
+  public static final String ICEBERG_BINARY_TYPE_ATTRIBUTE = "iceberg.binary-type";
+  public static final String ICEBERG_LONG_TYPE_ATTRIBUTE = "iceberg.long-type";
+  public static final String ICEBERG_FIELD_LENGTH = "iceberg.length";
 
   private static final ImmutableMap<Type.TypeID, TypeDescription.Category> TYPE_MAPPING =
       ImmutableMap.<Type.TypeID, TypeDescription.Category>builder()
           .put(Type.TypeID.BOOLEAN, TypeDescription.Category.BOOLEAN)
           .put(Type.TypeID.INTEGER, TypeDescription.Category.INT)
-          .put(Type.TypeID.TIME, TypeDescription.Category.INT)
           .put(Type.TypeID.LONG, TypeDescription.Category.LONG)
+          .put(Type.TypeID.TIME, TypeDescription.Category.LONG)
           .put(Type.TypeID.FLOAT, TypeDescription.Category.FLOAT)
           .put(Type.TypeID.DOUBLE, TypeDescription.Category.DOUBLE)
           .put(Type.TypeID.DATE, TypeDescription.Category.DATE)
-          .put(Type.TypeID.TIMESTAMP, TypeDescription.Category.TIMESTAMP)
```

Comment:
What's `Type.TypeID.TIMESTAMP` now mapping to?

Reply:
`Type.TypeID.TIMESTAMP` now can map to either `TypeDescription.Category.TIMESTAMP` or `TypeDescription.Category.TIMESTAMP_INSTANT` depending on with/without zone. The code for the same is present in `isSameType` where this map is being used.